### ISSUE TITLE
Segmentation fault with PHPUnit and Xdebug

### ISSUE THREAD

**Issue by hugoboos** [silexphp/Silex#1119]
Hi,

Take the following app.

```php
<?php
# src/app.php
require_once __DIR__ . "/../vendor/autoload.php";

$app = new Silex\Application();
$app->register(new \Silex\Provider\MonologServiceProvider());

$app['monolog'] = $app->share($app->extend("monolog", function($monolog, $app) {
  // Code
}));

$app->error(function (\Exception $exception) use ($app) {
  // Code
});

return $app;
```

```php
<?php
# tests/Test.php
class Test extends \Silex\WebTestCase
{
    public function createApplication()
    {
        return require __DIR__ . "/../src/app.php";
    }

    public function testHealthPage()
    {
        $client = $this->createClient();
        $client->request("GET", "/");
    }
}
```

Running the test with PHPUnit with code coverage enabled, will cause a segmentation fault.
```
$ vendor/bin/phpunit
PHPUnit 4.4.5 by Sebastian Bergmann.

Configuration read from /Users/hugoboos/Projects/Silex Xdebug/phpunit.xml

[1]    75447 segmentation fault  vendor/bin/phpunit
```

When you don't register the ```MonologServiceProvider```, extend ```$app['monolog']``` or register the error handler, everything is fine.

I've this problem with Xdebug 2.3.0 and 2.3.1. Not with Xdebug 2.2.7.

I'm on OS X 10.10.2, with PHP 5.6.6 (via Brew).

Any ideas?

**Comment by indeyets** [comment]
should be [reported to xdebug](http://bugs.xdebug.org/my_view_page.php), I believe

**Comment by hugoboos** [comment]
Bug report created at Xdebug: http://bugs.xdebug.org/view.php?id=1122

**Comment by naxhh** [comment]
Bumping this.

Seems was fixed in xdebug: 2.3.2

Should we close this?

**Comment by hugoboos** [comment]
Correct, it is fixed. Can be closed.
